cmd/monitor/monitor.go

```diff
@@ -106,6 +122,75 @@ func getChainState(ctx context.Context, ec *ethclient.Client) (*chainState, erro
 
 }
 
+func prependLatestBlocks(ctx context.Context, ms *monitorStatus, rpc *ethrpc.Client) {
```

Comment:
refractoring to make life easier hopefully. tried to comment the messy conditionals so you don't have to guess the code